File: backend/app/schemas/news.py
from datetime import datetime

from pydantic import BaseModel


# --- User-facing response models ---


class NewsArticleResponse(BaseModel):
    id: str
    source: str
    title: str
    url: str
    description: str | None
    audio_ready: bool
    audio_quality: str | None  # "parkiet" | "piper" | None
    published_at: datetime
    rendered_at: datetime | None


class NewsTodayResponse(BaseModel):
    date: str
    articles: list[NewsArticleResponse]
    total: int
    audio_ready_count: int


class NewsPreferencesRequest(BaseModel):
    feeds: list[str]
    max_articles: int = 20
    categories_exclude: list[str] = []


class NewsPreferencesResponse(BaseModel):
    feeds: list[str]
    max_articles: int
    categories_exclude: list[str]


# --- n8n ingest models ---


class NewsArticleCreate(BaseModel):
    source: str
    title: str
    url: str
    description: str = ""
    published_at: datetime

File: frontend-v2/src/lib/types/news.ts
export type NewsArticle = {
	id: string;
	source: string;
	title: string;
	url: string;
	description: string | null;
	audio_ready: boolean;
	audio_quality: 'parkiet' | 'piper' | null;
	published_at: string;
	rendered_at: string | null;
};

export type NewsTodayResponse = {
	date: string;
	articles: NewsArticle[];
	total: number;
	audio_ready_count: number;
};

export type NewsPreferences = {
	feeds: string[];
	max_articles: number;
	categories_exclude: string[];
};
